Sources/DNSMessage/Name.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+import NIOCore
+
+public struct DNSName: CustomStringConvertible, Hashable, Sendable, Equatable {
+    public internal(set) var labels: [DNSLabel]
+
+    public var description: String {
+        self.labels.map({ String(describing: $0) }).joined(separator: ".")
+    }
+
+    /// Total length in bytes including labels, separators, and terminating dot.
+    public var length: Int {
+        self.labels.reduce(0, { $0 + $1.bytes.count }) + self.labels.count + 1
+    }
+
+    /// Number of labels that make up this name
+    public var count: Int {
```

Comment:
Would `labelCount` make more sense? If I grab for `.count` on a string-like type I'd expect a character count